pkgs/by-name/ep/epson-escpr/package.nix

```
@@ -12,15 +19,27 @@ in stdenv.mkDerivation {
     # version.
     # NOTE: Don't forget to update the webarchive link too!
     urls = [
-      "https://download3.ebz.epson.net/dsc/f/03/00/13/76/45/5ac2ea8f9cf94a48abd64afd0f967f98c4fc24aa/epson-inkjet-printer-escpr-${version}-1lsb3.2.tar.gz"
+      "https://download3.ebz.epson.net/dsc/f/03/00/16/21/81/74d098a47c3a616713079c9cd5904b468bb33dea/epson-inkjet-printer-escpr-${version}.tar.gz"
 
-      "https://web.archive.org/web/https://download3.ebz.epson.net/dsc/f/03/00/13/76/45/5ac2ea8f9cf94a48abd64afd0f967f98c4fc24aa/epson-inkjet-printer-escpr-${version}-1lsb3.2.tar.gz"
+      "https://web.archive.org/web/https://download3.ebz.epson.net/dsc/f/03/00/16/21/81/74d098a47c3a616713079c9cd5904b468bb33dea/epson-inkjet-printer-escpr-${version}.tar.gz"
     ];
-    sha256 = "sha256:09rscpm557dgaflylr93wcwmyn6fnvr8nc77abwnq97r6hxwrkhk";
+    sha256 = "sha256-hVbX4OXPe4y37Szju3uVdXlVdjX4DFSN/A2Emz3eCcQ=";
   };
 
-  patches = [ ./cups-filter-ppd-dirs.patch ];
+  patches = [
+    ./cups-filter-ppd-dirs.patch
+    # DOS Line Endings in /lib directory and UNIX Line endings in /src directory.
+    # So a separate patch for /lib and /src
+    ./1.8.6-warnings-lib.patch
+    ./1.8.6-warnings-src.patch
+    ./1.8-missing-include.patch
```

```suggestion
    (fetchpatch {
      url = "https://gitweb.gentoo.org/repo/gentoo.git/plain/net-print/epson-inkjet-printer-escpr/files/1.8.6-warnings.patch";
      hash = lib.fakeHash;
    })
    ...
```
If those patches are all from Gentoo, please use [`fetchpatch`](https://nixos.org/manual/nixpkgs/unstable/#fetchpatch) instead of copying them in-tree.

---

This fails because of different line-endings error in patchPhase
```nix
    (fetchpatch {
        url="https://gitweb.gentoo.org/repo/gentoo.git/plain/net-print/epson-inkjet-printer-escpr/files/epson-inkjet-printer-escpr-1.8-missing-include.patch";
        hash="sha256-L4WhaxPQnJYyqCH00wiKIlFNMmCEXoGe5d7hJ5TMyEI=";
      })
    (fetchpatch {
      url = "https://gitweb.gentoo.org/repo/gentoo.git/plain/net-print/epson-inkjet-printer-escpr/files/1.8.6-warnings.patch";
      hash = "sha256-wSY0LZv2b+1kF7TfPolt554g79y2Ce6N/JqgjJyd3Ag=";
    })
```

```
@nix { "action": "setPhase", "phase": "patchPhase" }
applying patch /nix/store/x5gfirhz3sl53qf12cx38hp7ymxykjp6-cups-filter-ppd-dirs.patch
patching file configure
applying patch /nix/store/5cmxn3w61k1dk33x672vcpxczc6fkzwi-epson-inkjet-printer-escpr-1.8-missing-include.patch
patching file lib/epson-escpage.h
applying patch /nix/store/v72wz3lbds19xhqjv9gwkh94biliq43b-1.8.6-warnings.patch
patching file lib/epson-escpr-api.h
Hunk #1 FAILED at 111 (different line endings).
1 out of 1 hunk FAILED -- saving rejects to file lib/epson-escpr-api.h.rej
patching file src/filter.c
patching file src/mem.c
patching file src/wrapper.c
```

---

The `-l` flag might fix this depending on where the line ending is in the patch file, but it might be better to use the `decode` attr to pipe the patch through `dos2unix` (or `unix2dos`)

---

The issue is that the [warnings patch](https://gitweb.gentoo.org/repo/gentoo.git/tree/net-print/epson-inkjet-printer-escpr/files/1.8.6-warnings.patch) patches files in /lib with CRLF and files in /src with LF in a single patch file. 
So I have to do unix2dos on only a part of the file, which I do not know how to do with nix

---

Oh. That's... weird. Is `fetchpatch` converting Gentoo's correct line endings then? If so, then it can still be fetched with `fetchurl`.

---

No, gentoo is using dos2unix 
https://gitweb.gentoo.org/repo/gentoo.git/tree/net-print/epson-inkjet-printer-escpr/epson-inkjet-printer-escpr-1.8.6.ebuild#n31

---

I won't block this further, but it might be a good idea to do the same in a `postUnpack` hook.

---

Yeah I'm trying that thing out.

---

Done